src/dialogs/screenshot/screenshot.vala

```diff
@@ -0,0 +1,1322 @@
+
+		 public unowned Gdk.Monitor gdk_monitor {get; private set; }
+		 public int scale { get; private set; }
+
+		 public WaylandClient() {
+			 if (primary_monitor != null) return;
+			 libxfce4windowing.Screen.get_default().monitors_changed.connect(on_monitors_changed);
+			 on_monitors_changed();
+		 }
+
+		 void on_monitors_changed() {
+			 int loop = 0;
+
+			 /* it can take a short-time after first call for the underlying wayland client
+				to return a reference, so lets loop until we get a reference ... but
+				don't try indefinitely
+			 */
+			 Timeout.add(200, ()=> {
+				 primary_monitor = libxfce4windowing.Screen.get_default().get_primary_monitor();
+				 if (primary_monitor != null || loop++ > 10) {
+					 gdk_monitor = primary_monitor.get_gdk_monitor();
+					 scale = (int)primary_monitor.get_scale();
+					 return false;
+				 }
+
+				 return true;
+			 });
+		 }
+	 }
+
+	[SingleInstance]
+	public class CurrentState : GLib.Object {
+
+		GLib.Settings? buttonplacement;
+
+		public GLib.Settings screenshot_settings {get; private set;}
+		public int newstate {get; private set; default = WindowState.NONE;}
+		public bool showtooltips {get; set; default = true;}
+		public bool emit_capture_sound { get; set; default = true; }
+		public int buttonpos {get; private set;}
+		public string tempfile_path {get; private set;}
+		public bool startedfromgui {get; private set; default = false;}
+
+		public signal void buttonpos_changed();
+
+		public CurrentState() {
+			screenshot_settings = new GLib.Settings("org.buddiesofbudgie.budgie-desktop.screenshot");
+
+			buttonplacement = new GLib.Settings("com.solus-project.budgie-wm");
+
+			buttonplacement.changed["button-style"].connect(() => {
+				fill_buttonpos();
+				buttonpos_changed();
+			});
+			fill_buttonpos();
+
+			emit_capture_sound = screenshot_settings.get_boolean("screenshot-capture-sound");
+			screenshot_settings.changed["screenshot-capture-sound"].connect(() => {
+				emit_capture_sound = screenshot_settings.get_boolean("screenshot-capture-sound");
+			});
+
+			showtooltips = screenshot_settings.get_boolean("showtooltips");
+			screenshot_settings.changed["showtooltips"].connect(() => {
+				showtooltips = screenshot_settings.get_boolean("showtooltips");
+			});
+
+			// we need to use the same temporary user-space file across dbus client/server calls to coordinate
+			// the passing of screenshot images
+			string tmpdir = Environment.get_variable("XDG_RUNTIME_DIR") ?? Environment.get_variable("HOME");
+			tempfile_path = GLib.Path.build_path(GLib.Path.DIR_SEPARATOR_S, tmpdir, ".budgiescreenshot_tempfile");
+		}
+
+		private void fill_buttonpos() {
+			buttonpos = (buttonplacement.get_string("button-style") =="left") ? ButtonPlacement.LEFT : ButtonPlacement.RIGHT;
+		}
+
+		public void statechanged(int n) {
+			newstate = n;
+
+			(newstate == 0) ? startedfromgui = false : startedfromgui;
+			(newstate == 1) ? startedfromgui = true : startedfromgui;
```

Comment:
What are the magic numbers of 0 and 1? Can they be named, or at least commented?

Reply:
well found. tidied